src/vivarium/framework/results/manager.py

```diff
@@ -83,6 +83,14 @@ def on_post_setup(self, _: Event):
                         for stratification in self._results_context.stratifications
                         if stratification.name in stratification_names
                     ]
+                    missing_stratifications = set(stratification_names).difference(
+                        set(obs.name for obs in observation_stratifications)
+                    )
+                    if missing_stratifications:
+                        raise ValueError(
```

Comment:
Two notes:
* This will only record the missing stratifications from the first iteration that has some missing. It would be better to communicate all missing stratifications
* I think it' might be worth warning when stratifications are registered and then never used.